File: problem-solving/minimum-waiting-time/minimum.ts
export function minimumWaitingTime(queries: number[]): number {
  const xs = queries.sort((a, b) => a - b)
  let res = 0
  for (const [index, duration] of xs.entries()) {
    const queriesLeft = xs.length - (index + 1)
    // [1,2,2,3,6]
    // 4*1=1
    // 2*3 = 6
    // 2*2 = 4
    // 1 * 3 = 3
    // 0 * 0 = 0
    res += duration * queriesLeft
  }

  return res
}

// [3,2,1,2,6] // best = 17
// sorted [1,2,2,3,6]

const xs = [3, 2, 1, 2, 6]
console.log(minimumWaitingTime(xs))

File: problem-solving/minimum-waiting-time/minimum.go
package main

import (
	"fmt"
	"sort"
)

// MinimumWaitingTime Func
func MinimumWaitingTime(queries []int) int {
	xs := queries
	sort.Ints(xs)

	res := 0
	for i, v := range xs {
		queryTime := len(xs) - (i + 1)

		res += v * queryTime
	}
	return res
}

func main() {
	xs := []int{3, 2, 1, 2, 6}

	res := MinimumWaitingTime(xs)
	fmt.Println(res)
}
